pkg/services/factory.go

```diff
@@ -104,22 +105,14 @@ func (f *defaultFactory) GetKubeService() (kube.Kube, error) {
 }
 
 func (f *defaultFactory) GetGitClients(ctx context.Context, gpClient gitproviders.Client, params GitConfigParams) (git.Git, gitproviders.GitProvider, error) {
-	isExternalConfig := models.IsExternalConfigRepo(params.ConfigRepo)
-
-	var providerUrl string
-
-	switch {
-	case !params.IsHelmRepository:
-		providerUrl = params.URL
-	case isExternalConfig:
-		providerUrl = params.ConfigRepo
-	default:
-		return nil, nil, nil
+	// This is temporary. When config repo is always present in the config map this can be removed.
+	if params.ConfigRepo == "" && !params.IsHelmRepository {
+		params.ConfigRepo = params.URL
```

Comment:
At this point we might want to get rid of the field `ConfigRepo` and just leave URL.

Reply:
That is only in the case someone has the same config repo as the app they are adding. So running `gitops add app .` would hit this.